### ISSUE TITLE
Transactions are not getting mined

### ISSUE THREAD

**Issue by 314159265359879** [stacks-network/stacks-core#4185]
**Describe the bug**
Transaction that is well constructed does not get mined... not even after 1 day. Even though nonce is good, balance is available and in previous blocks STX transfers have been included with lower fees. 

This transaction was send 1 day ago:
https://explorer.hiro.so/txid/0x014ebde20c62582995aa4a4dcc854b47a5279650f911c988fcd6c5e26949aab8?chain=mainnet
![image](https://github.com/stacks-network/stacks-core/assets/33360391/638e56d7-6cf5-4ca0-af80-48164149d416)

Send from this address (last executed nonce is 15607)
![image](https://github.com/stacks-network/stacks-core/assets/33360391/754cb79b-7f05-4b18-8af1-272ec5538a9d)

More recent transactions from this address were sent with nonces that make no sense at all:
I think that may be why miners are now ignoring all the transactions pending and signed by this address.
![image](https://github.com/stacks-network/stacks-core/assets/33360391/07981f81-d01e-4ac2-b346-fa137bb4ac8a)

**Steps To Reproduce**
I am not sure this is reproducible, I haven't tried. I have been seeing situations like this since Stacks 2.1 and it pops up now and again. Here is another case, although somewhat different: https://github.com/stacks-network/stacks-core/issues/4118 (in 4118 there was a time when the pending transaction was unprocessable, here it looks like it was always processable). 

**Expected behavior**
We expect well-formed transactions to get mined, within a couple blocks, especially with a fee of 0.07 STX. 

**Environment (please complete the following information):**
n.a.

**Additional context**
Please include any relevant stack traces, error messages and logs.
Not sure how to make these.

**Comment by 314159265359879** [comment]
Update 20 min. ago the transaction got processed anyway, so those transactions that were pending for a day got processed after a day after all. Odd that it took that long. 

![image](https://github.com/stacks-network/stacks-core/assets/33360391/a771abae-dc35-4cf1-997f-0978f2169a2f)

**Comment by diwakergupta** [comment]
@314159265359879 so can we close this? If not, I'm not sure what the next steps would be. I briefly skimmed the logs for this txn on one of Hiro's nodes and nothing stood out - over last 2 days, there were just a handful of logs, all from 12/16 between 3:15 and 3:35 pm ET. Logs indicate normal transaction processing -- there are no indications that this transaction was ever received earlier and _not_ processed for any reason.

**Comment by obycode** [comment]
In Hiro's mock miner's logs, that transaction is first seen at 12/16/2023, 03:14:53.081, then is included in a block at 12/16/2023, 03:25:49.614. This seems to match up with the timing of when the tx was actually mined.

**Comment by 314159265359879** [comment]
@obycode those timestamps are GMT?

The mock miner mined the transaction 17 hours before it was mined for real?

I see this timestamp for the tx (CET, GMT+1)
![image](https://github.com/stacks-network/stacks-core/assets/33360391/b3a3d117-7910-495d-97c6-b4732ba7cd2a)

(Rounded up) for a day this transaction didn't reach a real miner or didn't get included despite the high fee. 

That is odd, right?

**Comment by obycode** [comment]
> @obycode those timestamps are GMT?

No, those logs were in EST, so they matched with when the tx was actually mined.

It's still unclear to me why that transaction didn't reach the mempool used by our nodes until then.

**Comment by shivaenigma** [comment]
Likely looks related to https://github.com/stacks-network/stacks-core/issues/4189#issuecomment-1896632692 . Miners are only picking up selective transactions leaving mempool congested
